src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/RemoveOrganizationUserCommand.cs

```diff
@@ -197,6 +201,12 @@ await _pushRegistrationService.DeleteUserRegistrationOrganizationAsync(devices,
     {
         var orgUsers = await _organizationUserRepository.GetManyAsync(organizationUsersId);
         var filteredUsers = orgUsers.Where(u => u.OrganizationId == organizationId).ToList();
+        var users = await _userRepository.GetManyAsync(filteredUsers.Select(u => u.Id));
+
+        if (users.Any(u => u.HasMasterPassword() == false))
+        {
+            throw new BadRequestException(RemoveTDEUsersErrorMessage);
+        }
```

Comment:
This command is a real jumble of code paths. But I don't think this is executed by all public methods - only the 2 "bulk" public methods.

I suggest
- moving to a new private method
- calling that private method from here (used by the 2 bulk methods)
- also calling it from `ValidateRemoveUser` (seems to be used by all other methods?)

But please check and make sure this covers all code paths.